client/src/context.js

```diff
@@ -0,0 +1,128 @@
+import React, { createContext, useContext, useReducer } from "react";
```

Comment:
General comment: since there are some complex workflows in this file, would it be helpful to add some comments? (Not a requirement -- just thinking there may be some orders of operations and caveats that are worth capturing while they're fresh in mind, so this is easier to understand when revisited in the future).

Reply:
Made change.